support/globalconfig/network.go

```diff
@@ -31,4 +31,16 @@ func ReconcileNetworkConfig(cfg *configv1.Network, hcp *hyperv1.HostedControlPla
 		cfg.Spec.ExternalIP = globalConfig.Network.Spec.ExternalIP
 		cfg.Spec.ServiceNodePortRange = globalConfig.Network.Spec.ServiceNodePortRange
 	}
+
+	// Without this, the CNOs proxy controller refuses to reconcile the proxy status
+	// The CNO only populates this after MTU probing and it is required for the proxy
+	// controller to populate proxy status. Proxy not being populated makes the MTU
+	// probing fail if there is a proxy. Get out of the deadlock by initially populating
+	// it if unset.
+	if len(cfg.Status.ClusterNetwork) == 0 {
+		cfg.Status.ClusterNetwork = cfg.Spec.ClusterNetwork
+	}
+	if len(cfg.Status.ServiceNetwork) == 0 {
+		cfg.Status.ServiceNetwork = cfg.Spec.ServiceNetwork
```

Comment:
how is it done in regular ocp?

Reply:
Read the explanation in the PR description and comment. The network controller does it but can not in Hypershift with proxy due to a deadlock situation.

Reply:
(that deadlock doesn't exist in self-hosted, because there you never need a proxy to get to the KAS)